e2e/app/component/export-to-notebook-modal.ts

```diff
@@ -0,0 +1,63 @@
+import RadioButton from 'app/element/radiobutton';
+import Textbox from 'app/element/textbox';
+import {Page} from 'puppeteer';
+import {savePageToFile, takeScreenshot} from 'utils/save-file-utils';
+import {Language, LinkText} from 'app/text-labels';
+import Modal from './modal';
+
+export default class ExportToNotebookModal extends Modal {
+   // data-test-id="notebook-name-input"
+
+  constructor(page: Page, xpath?: string) {
+    super(page, xpath);
+  }
+
+  async waitForLoad(): Promise<this> {
+    try {
+      await super.waitForLoad();
+      await (this.getNotebookNameInput()).asElementHandle();
+    } catch (e) {
+      await savePageToFile(this.page);
+      await takeScreenshot(this.page);
+      throw new Error(`ExportToNotebookModal waitForLoad() encountered ${e}`);
+    }
+    return this;
+  }
+
+  getNotebookNameInput(): Textbox {
+    const selector = `${this.getXpath()}//*[@data-test-id="notebook-name-input"]`;
+    return new Textbox(this.page, selector);
+  }
+
+  getPythonRadiobutton(): RadioButton {
+    const selector = `${this.getXpath()}//label[contains(normalize-space(),"Python")]//input[@type="radio"]`;
+    return new RadioButton(this.page, selector);
+  }
+
+  getRRadiobutton(): RadioButton {
+    const selector = `${this.getXpath()}//label[contains(normalize-space(),"R")]//input[@type="radio"]`;
+    return new RadioButton(this.page, selector);
+  }
+
+  /**
+   * Fill out Export Notebook modal to create a new notebook:
+   * - Type notebook name.
+   * - Select notebook language.
+   * - Click "Export and Open" button.
+   * @param {string} notebookName Notebook name.
+   * @param {Language} language Notebook programming language. Default value is Python.
+   */
+  async fillInModal(notebookName: string, language: Language = Language.Python): Promise<void> {
+    const notebookNameInput = this.getNotebookNameInput();
+    await notebookNameInput.type(notebookName);
+    if (language === Language.Python) {
+      const radio = this.getPythonRadiobutton();
+      await radio.select();
+    } else {
+      const radio = this.getRRadiobutton();
+      await radio.select();
+    }
```

Comment:
👍 Done

Reply:
This will break if we add a third language; I think it's important to distinguish a two-state enum from a boolean. I.e. I'd want to see a switch on the language with a default to throw an unrecognized language error.